src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int size;
+    private Object[] elementData;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_SIZE];
+
+    }
+
+    public void checkCapacity() {
+        if (size == elementData.length) {
+            int newSize = (Integer) elementData.length * 3 / 2;
+            Object[] newArray = new Object[newSize];
+            System.arraycopy(elementData, 0, newArray, 0, size);
+            elementData = newArray;
+        }
+    }
+
     @Override
     public void add(T value) {
+        checkCapacity();
+        elementData[size++] = value;
 
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForSet(index);
+        checkCapacity();
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
 
     }
 
+    public void checkIndexForSet(int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+    }
+
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+
+        }
 
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elementData[index];
+
+    }
+
+    public void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
```

Comment:
The exception `ArrayListIndexOutOfBoundsException` is not a standard Java exception. Consider using `IndexOutOfBoundsException` instead.